### ISSUE TITLE
Easier Access Token Retrieval via React Hook

### ISSUE THREAD

**Issue by olafwrieden** [AzureAD/microsoft-authentication-library-for-js#4232]
### Core Library

MSAL.js v2 (@azure/msal-browser)

### Wrapper Library

MSAL React (@azure/msal-react)

### TL;DR
I would like to see a better way of retrieving a valid access token from MSAL that doesn't involve taking up half of my file (yes, I'm looking at you `useEffect()` hooks). See proposed changes below.

---

### Scenario
React.js page loads, it needs some data fetched from my API so I use a `useEffect()` hook to make the API call on page mount. Because my API endpoints are protected with bearer token auth, I need to pass along a the access token given to me by msal-react on the frontend.

### Problem
Retrieving the access token on the frontend just so that I can pass it along to my generic API call function is a *MASSIVE* pain. It requires (as per documentation) around [23 lines of code](https://github.com/Azure-Samples/ms-identity-javascript-react-tutorial/blob/main/3-Authorization-II/1-call-api/SPA/src/pages/Hello2.jsx#L21-L43) in every single file just to acquire the token used to make an API call in single line.

### Proposed Changes
I would like to see a more efficient way of retrieving access tokens that abstracts the logic around getting the access token silently or forcing a popup/redirect to login OR improving the documentation to show how this can be done in one place, rather than in every single `useEffect()` hook where data needs to be fetched from an authenticated API (could someone please share/comment if they have managed to do this this neatly).

Auth0 provides a great example of a `getAccessTokenSilently` function as is [documented here](https://auth0.com/docs/quickstart/spa/react/02-calling-an-api#get-an-access-token). Something like this, **with built-in verbose fallback to popup or redirect auth when the silent function fails**, would be a welcome addition to the `useMsal()` hook and allow for a huge reduction of repetitive code inside the `useEffect()` hooks we build today:

```javascript
const accessToken = await getAccessTokenSilently({
    audience: `api://${domain}`,
    scope: "User.Read",
});
```

### Source

External (Customer)

**Comment by tnorling** [comment]
Thanks this is something we've discussed and we have an item in our backlog to look into this. For now, you can call `acquireTokenSilent` inside your generic API function if you'd like. You can even call your interactive methods inside your generic API function as long as that function is **only** called from inside a react component. Our [sample](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/samples/msal-react-samples/react-router-sample/src/utils/MsGraphApiCall.js) and [getting started doc](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-react/docs/getting-started.md#acquiring-an-access-token-outside-of-a-react-component) demonstrates calling `acquireTokenSilent` from the function that makes the API call. Alternatively, you may consider writing your own hook to reuse this logic.
